### PR TITLE
Move templates from OTX1.X to OTX2.X

### PR DESCRIPTION
**Pull request by kprokofi**
### Summary
Move templates from OTX1.6 to OTX2.2 to preserve contract between Geti and OTX and to be able to control hyperparameters from OTX2.X side

This is a workaround. In the future we will migrate to our OTX2.X recipes.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/recipe/_base_/train.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ callbacks:
       check_on_train_epoch_end: false
       min_delta: 0.001
       warmup_iters: 30
-      warmup_epochs: 3
+      warmup_epochs: 7
```
Comment (sungchul1) [inline src/otx/recipe/_base_/train.yaml]: Why is this update needed?
And I think this change must affect other tasks' performance, could you check performance difference before and after this update?

File: src/otx/tools/templates/detection/detection/configuration.yaml [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+    enum_name: InputSizePreset
+    header: Configure model input size.
+    options:
+      DEFAULT: "Default"
+      AUTO: "Auto"
+      _256x256: "256x256"
+      _384x384: "384x384"
+      _512x512: "512x512"
+      _768x768: "768x768"
+      _1024x1024: "1024x1024"
```
Comment (sungchul1) [inline src/otx/tools/templates/detection/detection/configuration.yaml:264]: I don't know much about this, aren't other input sizes needed, like (640, 640) for yolox or (800, 992) for atss?

File: src/otx/recipe/_base_/train.yaml (hunk 1 above)
Comment (harimkang) [reply src/otx/recipe/_base_/train.yaml]: Yes in my opinion there shouldn't be any modifications to `/_base_/train.yaml`. If we need this, we need to update this in each model recipe.

File: src/otx/tools/templates/detection/detection/configuration.yaml (hunk 2 above)
Comment (harimkang) [reply src/otx/tools/templates/detection/detection/configuration.yaml:264]: This is a copy of OTX 1.6, which is set to invisible in the Geti UI, so only Default is used for Geti. If this needs to be visible in the UI in the future, it might be worth revisiting, but this is just a temporary file for now.

File: src/otx/tools/templates/classification/mobilenet_v3_large_1_cls_incr/template.yaml [hunk 3]
```diff
@@ -26,7 +26,7 @@ hyper_parameters:
         default_value: 0.0058
         auto_hpo_state: POSSIBLE
       learning_rate_warmup_iters:
-        default_value: 100
+        default_value: 10
```
Comment (kprokofi) [inline src/otx/tools/templates/classification/mobilenet_v3_large_1_cls_incr/template.yaml]: @harimkang , please note, that in OTX2.2 it is not the epochs, but iterations. So, the learning rate will be warmed up only 10 iterations. I think it is a bit useless.

File: src/otx/tools/templates/detection/detection/cspdarknet_yolox_l/template.yaml [hunk 4]
```diff
@@ -28,7 +28,7 @@ hyper_parameters:
         default_value: 0.001
         auto_hpo_state: POSSIBLE
       learning_rate_warmup_iters:
-        default_value: 100
+        default_value: 3
```
Comment (kprokofi) [inline src/otx/tools/templates/detection/detection/cspdarknet_yolox_l/template.yaml:31]: @sungchul2 , it is not epochs. It is iterations. 3 iterations of warm-up is useless. Please, consider higher value

File: src/otx/recipe/_base_/train.yaml (hunk 1 above)
Comment (kprokofi) [reply src/otx/recipe/_base_/train.yaml]: Okay, I will revert it, but to be honest, 7-10 epochs are reasonable value to start early stopping, 3 epochs are not enough for many models to stabilize on small datasets

File: src/otx/tools/templates/classification/mobilenet_v3_large_1_cls_incr/template.yaml (hunk 3 above)
Comment (harimkang) [reply src/otx/tools/templates/classification/mobilenet_v3_large_1_cls_incr/template.yaml]: Yeah.. 
As with the other models, set it to 0 to disable it for now. https://github.com/openvinotoolkit/training_extensions/pull/3951/commits/5ca8b4341248bfd33142504c0de071f75770c546

File: src/otx/recipe/_base_/train.yaml (hunk 1 above)
Comment (kprokofi) [reply src/otx/recipe/_base_/train.yaml]: reverted https://github.com/openvinotoolkit/training_extensions/pull/3951/commits/d3e910e76cf62f2f63f8ea5623041c6bafc4e9f9

Comment (sungchul1) [reply src/otx/recipe/_base_/train.yaml]: We cannot be sure until we test it experimentally because small epochs may be enough on small datasets because models only have to adapt to the distribution for a small number of samples.
But this is not in the scope of this PR that temporarily adds templates.

File: src/otx/tools/templates/detection/detection/cspdarknet_yolox_l/template.yaml (hunk 4 above)
Comment (sungchul1) [reply src/otx/tools/templates/detection/detection/cspdarknet_yolox_l/template.yaml:31]: I know and I think it is strange setting as well :D
But it is just for following recipes.

Comment (kprokofi) [reply src/otx/tools/templates/detection/detection/cspdarknet_yolox_l/template.yaml:31]: Do we have also iterations for this parameter in OTX1.6? Or maybe it was epochs

Comment (sungchul1) [reply src/otx/tools/templates/detection/detection/cspdarknet_yolox_l/template.yaml:31]: I'm not sure but maybe iteration. @harimkang do you have any idea?

Comment (harimkang) [reply src/otx/tools/templates/detection/detection/cspdarknet_yolox_l/template.yaml:31]: But this is actually going to be converted to an OTX 2.2 recipe now, so we can use it the way we're using it now, right? (We just need to choose whether we want to use epochs or iter, and it's up to the IAI team to change this key to iter -> epoch.)

Comment (kprokofi) [reply src/otx/tools/templates/detection/detection/cspdarknet_yolox_l/template.yaml:31]: Yes, let's consider hyperparameters update, especially for warm-up and early stopping in the nearest future.

**Review by eugene123tw — COMMENTED** [review]
Thanks, Kirill! I have a few general questions:

* Even though new models aren’t introduced in OTX 1.x, should we still include them in `converter.py`? For example, we didn’t have `Custom_Instance_Segmentation_RTMDet_tiny` in OTX 1.x, but it's still present in the converter. I assume this is more about ensuring Geti recognizes the format rather than converting from OTX 1.x to OTX 2.x. Is that correct?

**Comment by kprokofi** [comment]
> Thanks, Kirill! I have a few general questions:
> 
> * Even though new models aren’t introduced in OTX 1.x, should we still include them in `converter.py`? For example, we didn’t have `Custom_Instance_Segmentation_RTMDet_tiny` in OTX 1.x, but it's still present in the converter. I assume this is more about ensuring Geti recognizes the format rather than converting from OTX 1.x to OTX 2.x. Is that correct?

Actually, we added new model templates and updated converter accordingly. RTMDet model will be added in Geti new version. Yunchu will update docker image and OTX2.2 will have separated templates from 1.6. So, adding new models are not a problem.
The main problem today is that all components from Geti are relied on the old contract between OTX and Geti. So, we cannot change the mapping between new recipes and Geti fast. IAI team will adapt their side for our new recipes, but it requires some time and for now we need to preserve old templates.

**Review by harimkang — APPROVED** [review]

**Review by sungchul1 — APPROVED** [review]
